internal/helm/chart/builder_remote.go

```diff
@@ -171,6 +152,121 @@ func (b *remoteChartBuilder) Build(_ context.Context, ref Reference, p string, o
 	return result, nil
 }
 
+func (b *remoteChartBuilder) downloadFromOCIRepository(remote *repository.OCIChartRepository, remoteRef RemoteReference, buildResult *Build, opts BuildOptions) (*bytes.Buffer, error) {
```

Comment:
The function arguments can be simplified by removing `remote` repository to be passed, as `remoteChartBuilder` embeds the `remote`. If the interface is expanded as suggested above, `Get()`, `Unload()` and `DownloadChart()` can be directly called on the `remote` without the need of any type casting.
If the remote interface is not expanded to have `Unload()`, as this function is aware of when it's called, instead of passing a typed remote, it can internally do the type casting. I think expanding the remote interface would help not have two separate functions for downloading, as most of the code is exactly the same.

Also, the `buildResult` argument can just become a returned value. Don't see any specific reason to accept it as a pointer. Whenever there's an error and `buildResult` is not assigned, the caller always ends further building and stops any usage of `buildResult`.

All these comments apply to `downloadFromRepository()` as well.

Reply:
Now that I read this, an `Unload()` method in the interface makes more sense 👍